helix-term/src/commands/typed.rs

```diff
@@ -1445,12 +1449,13 @@ pub(super) fn goto_line_number(
             let text = doc.text().slice(..);
             let line = doc.selection(view.id).primary().cursor_line(text);
             cx.editor.last_line_number.get_or_insert(line + 1);
```

Comment:
It might be a good time to refactor this too. The line number is currently computed with every update but it is only needed when last_line_number is None.  Using `get_or_insert_with(|doc, view| ...`  instead, it can be computed when needed.